### PR TITLE
refactor(AutoComplete): refactor autocomplete and add delete documents

### PR DESCRIPTION
**Pull request by zhenlei520**

### CODE REVIEW

File: src/SearchEngine/Masa.BuildingBlocks.SearchEngine.AutoComplete/BaseAutoCompleteClient.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
+﻿namespace Masa.BuildingBlocks.SearchEngine.AutoComplete;
+
+public abstract class BaseAutoCompleteClient : IAutoCompleteClient
```
Comment (doddgu) [inline src/SearchEngine/Masa.BuildingBlocks.SearchEngine.AutoComplete/BaseAutoCompleteClient.cs:3]: To be confirm BaseXXX or XXXBase?

File: src/SearchEngine/Masa.BuildingBlocks.SearchEngine.AutoComplete/BaseAutoCompleteClient.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    public abstract Task<GetResponse<TAudoCompleteDocument, TValue>> GetAsync<TAudoCompleteDocument, TValue>(string keyword,
+        AutoCompleteOptions? options = null,
+        CancellationToken cancellationToken = default) where TAudoCompleteDocument : AutoCompleteDocument<TValue> where TValue : notnull;
+
+    public virtual Task<SetResponse> SetAsync(AutoCompleteDocument<Guid> document, SetOptions? options = null,
+        CancellationToken cancellationToken = default)
+        => SetAsync<AutoCompleteDocument<Guid>, Guid>(document, options, cancellationToken);
+
+    public virtual Task<SetResponse> SetMultiAsync(IEnumerable<AutoCompleteDocument<Guid>> documents, SetOptions? options = null,
```
Comment (doddgu) [inline src/SearchEngine/Masa.BuildingBlocks.SearchEngine.AutoComplete/BaseAutoCompleteClient.cs:22]: SetMulti? SetList? or Set?

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/SearchEngine/Masa.BuildingBlocks.SearchEngine.AutoComplete/BaseAutoCompleteClient.cs (hunk 1 above)
Comment (zhenlei520) [reply src/SearchEngine/Masa.BuildingBlocks.SearchEngine.AutoComplete/BaseAutoCompleteClient.cs:3]: It has been confirmed, it is BaseXXX

File: src/SearchEngine/Masa.BuildingBlocks.SearchEngine.AutoComplete/BaseAutoCompleteClient.cs (hunk 2 above)
Comment (zhenlei520) [reply src/SearchEngine/Masa.BuildingBlocks.SearchEngine.AutoComplete/BaseAutoCompleteClient.cs:22]: In order to make single and multiple operations clearer, it is divided into two. It is more suitable to look at SetList here, but it is a bit strange to put it into delete operation. IRepository uses Range, but it doesn't feel good to use SetRange here.

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**